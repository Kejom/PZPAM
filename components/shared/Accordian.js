import { Pressable, StyleSheet, Text, View, LayoutAnimation, Platform, UIManager } from "react-native";
import {MaterialIcons} from "@expo/vector-icons";
import { useState } from "react";
import { GlobalColors } from "../../constants/colors";

export default function Accordian({children, title, style}){
    const [expanded, setExpanded] = useState(false);

    if(Platform.OS === 'android' && UIManager.setLayoutAnimationEnabledExperimental)
        UIManager.setLayoutAnimationEnabledExperimental(true);

    function toggleExpanded(){
        LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut);
        setExpanded(state => !state);
    }
    return (
        <View style={style}>
            <Pressable onPress={toggleExpanded}>
                <View style={styles.row}>
                    <Text style={styles.title}>{title}</Text>
                    <MaterialIcons name={expanded ? 'keyboard-arrow-up': 'keyboard-arrow-down'} size={30} color="white"/>
                </View>
            </Pressable>
            {expanded && <View style={styles.childContainer}>{children}</View>}
        </View>
    )
}

const styles = StyleSheet.create({
    row: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        height: 56,
        paddingLeft: 25,
        paddingRight: 18,
        alignItems: 'center',
        backgroundColor: GlobalColors.primaryDark,
    },
    title: {
        fontSize: 16,
        fontWeight: 'bold',
        color: 'white'
    },
    childContainer: {
        backgroundColor: GlobalColors.primary,
        borderWidth: 1,
        borderColor: GlobalColors.greyDark
    }
})