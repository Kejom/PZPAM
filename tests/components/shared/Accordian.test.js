import React from 'react';
import { render} from '@testing-library/react-native';
import Accordian from '../../../components/shared/Accordian';
import { Text } from 'react-native';

describe('Accordian', () => {
    it('renders correctly', () => {
        const { toJSON } = render(
          <Accordian title="Title">
            <Text>Content</Text>
          </Accordian>
        );
        expect(toJSON()).toMatchSnapshot();
      });
  });