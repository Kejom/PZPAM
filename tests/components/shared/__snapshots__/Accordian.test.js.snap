// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordian renders correctly 1`] = `
<View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#2B702D",
          "flexDirection": "row",
          "height": 56,
          "justifyContent": "space-between",
          "paddingLeft": 25,
          "paddingRight": 18,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Title
      </Text>
      <Text />
    </View>
  </View>
</View>
`;
